src/main/java/org/folio/circulation/domain/LocationRepository.java

```diff
@@ -168,4 +170,31 @@ public static LocationRepository using(Clients clients) {
         institution.getString("id"))));
   }
 
+  public CompletableFuture<Result<Collection<Location>>> loadLibraries(Collection<Location> locations) {
+    return getLibraries(locations)
+      .thenApply(flatMapResult(libraries -> succeeded(
+        locations.stream()
+          .map(location -> location.withLibraryRepresentation(
+            libraries.getOrDefault(location.getLibraryId(), null)))
+          .collect(toSet()))));
+  }
+
+  public CompletableFuture<Result<Collection<Location>>> loadInstitutions(Collection<Location> locations) {
+    return getInstitutions(locations)
+      .thenApply(flatMapResult(institutions -> succeeded(
+        locations.stream()
+          .map(location -> location.withInstitutionRepresentation(
+            institutions.getOrDefault(location.getInstitutionId(), null)))
+          .collect(toSet()))));
+  }
+
+  public CompletableFuture<Result<Collection<Location>>> loadCampuses(Collection<Location> locations) {
+    return getCampuses(locations)
+      .thenApply(flatMapResult(campuses -> succeeded(
+        locations.stream()
+          .map(location -> location.withCampusRepresentation(
+            campuses.getOrDefault(location.getCampusId(), null)))
+          .collect(toSet()))));
+  }
```

Comment:
Do these need to be public? Is the hierarchy always fetched for a location?

Reply:
The do, if we want to call them from another package, and we do. No, not always.

Reply:
Ok. Please can they be named to be more consistent with the names of other public repository methods e.g. find or fetch, rather than load.

Reply:
done